scheds/rust/scx_lavd/src/bpf/main.bpf.c

```diff
@@ -2019,7 +2074,32 @@ void BPF_STRUCT_OPS(lavd_dispatch, s32 cpu, struct task_struct *prev)
 		bpf_task_release(p);
 		break;
 	}
+
+	if (no_2_level_scheduling || ret)
+		goto unlock_out;
   
+	bpf_for_each(scx_dsq, p, LAVD_REGULAR_DSQ, 0) {
+		if (is_kernel_task(p)) {
```

Comment:
This is more a question about sched_ext in general, since we have the same behavior pretty much with all the schedulers, but I'll just ask here.

What happens if the task can't run on the CPU where it's consumed (e.g., if `cpu` is not allowed according to  `p->cpus_ptr)`? Is it just re-enqueued? If that's the case I was wondering if it'd be more efficient to check here if cpu is allowed before consuming the task, otherwise we could move to the next task and kick `scx_bpf_task_cpu(p)`. WDYT?

Reply:
As far as I understand, `scx_bpf_consume` iterates the specified DSQ and consumes the first consumable task, respecting task's cpumask. If there is no such task, it will return false.

Reply:
You're right, this check is implemented in `task_can_run_on_remote_rq()`. Hm... I'm wondering how much overhead tasks that can only run on certain CPUs (or even worse only on 1 CPU) can introduce  when they are dispatched to a global DSQ, Would it be more efficient to dispatch them to a per-CPU DSQ? This would allow to save some cycles when the global DSQ is consumed from all the CPUs, but I'm not sure if it's worth it, considering that we may also need to avoid prioritizing these tasks too much... need to do some tests with this.

Reply:
It would obviously depend on the frequency but skipping over tasks in DSQ isn't particularly expensive and there usually aren't *that* many tasks with modified cpumasks. I doubt the overhead would be noticeable especially on smaller machines.